tests/zephyr/drivers/flash/common/boards/nrf54l15dk_nrf54l15_cpuapp_sqspi.overlay

```diff
@@ -0,0 +1,100 @@
+&pinctrl {
+	sqspi_default: sqspi_default {
+		group1 {
+			psels = <NRF_PSEL(SDP_MSPI_SCK, 2, 1)>,
+				<NRF_PSEL(SDP_MSPI_CS0, 2, 5)>,
+				<NRF_PSEL(SDP_MSPI_DQ0, 2, 2)>;
+			nordic,drive-mode = <NRF_DRIVE_E0E1>;
+			bias-pull-up;
+		};
+		group2 {
+			psels = <NRF_PSEL(SDP_MSPI_DQ1, 2, 4)>,
+				<NRF_PSEL(SDP_MSPI_DQ2, 2, 3)>,
+				<NRF_PSEL(SDP_MSPI_DQ3, 2, 0)>;
+			nordic,drive-mode = <NRF_DRIVE_E0E1>;
+			bias-pull-up;
+		};
+	};
+
+	sqspi_sleep: sqspi_sleep {
+		group1 {
+			low-power-enable;
+			psels = <NRF_PSEL(SDP_MSPI_SCK, 2, 1)>,
+				<NRF_PSEL(SDP_MSPI_CS0, 2, 5)>,
+				<NRF_PSEL(SDP_MSPI_DQ0, 2, 2)>,
+				<NRF_PSEL(SDP_MSPI_DQ1, 2, 4)>,
+				<NRF_PSEL(SDP_MSPI_DQ2, 2, 3)>,
+				<NRF_PSEL(SDP_MSPI_DQ3, 2, 0)>;
+		};
+	};
+};
+
+&cpuflpr_vpr {
+	pinctrl-0 = <&sqspi_default>;
+	pinctrl-1 = <&sqspi_sleep>;
+	pinctrl-names = "default", "sleep";
+	interrupts = <76 NRF_DEFAULT_IRQ_PRIORITY>;
+	status = "okay";
+};
+
+/ {
+	reserved-memory {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		ranges;
+
+		softperipheral_ram: memory@2003c000 {
+			reg = <0x2003c000 0x4000>;
+			ranges = <0 0x2003c000 0x4000>;
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			sqspi: sqspi@3c00 {
+				compatible = "nordic,nrf-sqspi";
+				#address-cells = <1>;
+				#size-cells = <0>;
+				reg = <0x3c00 0x200>;
+				status = "okay";
+				zephyr,pm-device-runtime-auto;
+				/* This is needed because the flash_mspi_nor
+				 * driver is currently broken and does not
+				 * reconfigure properly the MSPI controller
+				 * when used in the single-peripheral mode.
+				 * To be removed when this is fixed.
+				 */
+				software-multiperipheral;
+			};
+		};
+	};
+};
+
+/delete-node/ &mx25r64;
+
+&sqspi {
+	mx25r64: mx25r6435f@0 {
+		compatible = "jedec,mspi-nor";
```

Comment:
Alignment similar to this:
https://github.com/nrfconnect/sdk-nrf/pull/21325/commits/958fa23ca38b2a83be7fb41468c47e5a9daf2d76
will be needed when #21325 is merged.